src/modules/user/middlewares/verifyPermission.js

```diff
@@ -0,0 +1,27 @@
+const { STATUS_CODES } = require('../../../../constants/constants');
+const AppError = require('../../../../utils/appErrors');
+const { getUserPermission } = require('../service/userService');
+
+const checkAccess = (requiredPermission) => async (req, res, next) => {
+  try {
+    const { userId } = req.user;
+
+    const permissionMapping = await getUserPermission(userId);
+    const hasRequiredPermission = permissionMapping.some(
```

Comment:
.some checks if atleast one condition works, whereas all permissions should be matched